### DISCUSSION TITLE
Overwriting i18n `en` text

### DISCUSSION THREAD

**Discussion by burbma** [mikker/passwordless#216] [Q&A]
Disclaimer: I'm new to Rails. I don't think this is a passwordless problem specifically but I'm having trouble finding help elsewhere. I discovered that the subject of the sign in email comes from this line https://github.com/mikker/passwordless/blob/master/config/locales/en.yml#L25. I wanted to customize it so I made a `config/locales/en.yml` file like so:
```yaml
en:
  passwordless:
    mailer:
      sign_in:
        subject: "Signing in to My Awesome App"
```
(I found I can also put the file at `config/locales/passwordless/en.yml`). When working in development locally I find that it uses my subject just fine. When I deploy it to Heroku however it uses the default subject from passwordless. I have a vanilla Rails 7 app via rails new and I haven't changed anything in development.rb or production.rb or application.rb. I don't see anything about the priority order of Rails.configuration.i18n.load_path being customizable or problematic in any of my searches. Any guidance anyone can provide would be appreciated.

**Comment by mikker** [comment]
Huh, that doesn't sound like anything I've heard before.

Are you sure…
* all the keys are spelled right and in the right cAsE?
* that the locale file is getting properly deployed?
* that your default locale is set to English?

**Comment by burbma** [comment]
Thanks for the response!

Are you sure…
* all the keys are spelled right and in the right cAsE?
    * I copy/pasted my yml right into my question so we see it's spelled right.
    * Also, the subject is overwritten in dev mode locally as I expect it to be. I don't think prod would be looking in a different place for the locale yml.
* that the locale file is getting properly deployed?
    * I think so as evidenced by my load_path comments in the next bullet, also when I `cat` the file on Heroku it looks correct, here it is
```
~ $ cat /app/config/locales/passwordless/en.yml
en:
  passwordless:
    mailer:
      sign_in:
        subject: "Signing in to My Awesome App"
```
* that your default locale is set to English?
    * I hadn't done anything to set default locales or any other locales besides English. I was under the impression that the fallback default was `en`. So I checked it out in the console. Both locally and on Heroku the `I18n.locale` and `I18n.default_locale` were `:en`. However, locally `I18n.available_locales` was `[:en, :da, :ru]` while on Heroku it was a huge list. So following [instructions in the rails guide](https://guides.rubyonrails.org/i18n.html#configure-the-i18n-module) I added

```ruby
    config.i18n.load_path += Dir[Rails.root.join('config', 'locales', '*.{rb,yml}')]
    config.i18n.load_path += Dir[Rails.root.join('config', 'locales', '**', '*.{rb,yml}')]
    config.i18n.available_locales = [:en]
    config.i18n.default_locale = :en
```
      
to my `config/application.rb` and now the configs I was looking at matched locally and on Heroku. However the sign_in email subject was still not working on Heroku. When I look at `I18n.load_paths` I see that locally my apps yml's are last in the array while on Heroku the passwordless gem yml is last in the array. This is interesting because of the appending to the load_path that I did in my config/application.rb. What's really interesting is that the passwordless gem yml files appear _twice_ in the Heroku deployment's `I18n.load_path`, once before my yml files (where it shows up locally as well) and once at the end. My appending to the load_path seems to cause my yml's to be in there more than once as well, but still before the second occurrence of the passwordless gem's yml. The rails guide does't seem mention if this load_paths array is specifying priority order or not.

**Reply by arinhouck** [reply]
@mikker I'm running into the same exact issue. However I'm deploying to AWS.

I have also tried setting defaults like above as well. I can fix the issue in production rails console if I run manually to load paths but cannot seem to fix at runtime of the application using application config or an initializer. 

Passwordless load path is injected at the very end in my production environment as well.

Locally everything is working fine.

EDIT: I found the open issue is merged now. I am going to pull down the latest version!
